tests/integration/test_failed_mutations/test.py

```diff
@@ -0,0 +1,125 @@
+import logging
+import random
+import threading
+import time
+from collections import Counter
+
+import pytest
+from helpers.cluster import ClickHouseCluster
+
+cluster = ClickHouseCluster(__file__)
+
+node_with_backoff = cluster.add_instance(
+    "node_with_backoff",
+    macros={"cluster": "test_cluster"},
+    main_configs=["configs/config.d/backoff_mutation_policy.xml"],
+    with_zookeeper=True,
+)
+
+node_no_backoff = cluster.add_instance(
+    "node_no_backoff",
+    macros={"cluster": "test_cluster"},
+    with_zookeeper=True,
+)
+
+REPLICATED_POSPONE_MUTATION_LOG = (
+    "According to exponential backoff policy, put aside this log entry"
+)
+POSPONE_MUTATION_LOG = (
+    "According to exponential backoff policy, do not perform mutations for the part"
+)
+
+all_nodes = [node_with_backoff, node_no_backoff]
+
+
+def prepare_cluster(use_replicated_table):
+    for node in all_nodes:
+        node.query("DROP TABLE IF EXISTS test_mutations SYNC")
+
+    engine = (
+        "ReplicatedMergeTree('/clickhouse/{cluster}/tables/test/test_mutations', '{instance}')"
+        if use_replicated_table
+        else "MergeTree()"
+    )
+
+    for node in all_nodes:
+        node.query(f"CREATE TABLE test_mutations(x UInt32) ENGINE {engine} ORDER BY x")
+        node.query("INSERT INTO test_mutations SELECT * FROM system.numbers LIMIT 10")
+
+
+@pytest.fixture(scope="module")
+def started_cluster():
+    try:
+        cluster.start()
+        yield cluster
+
+    finally:
+        cluster.shutdown()
+
+
+@pytest.mark.parametrize(
+    ("node, found_in_log"),
+    [
+        (
+            node_with_backoff,
+            True,
+        ),
+        (
+            node_no_backoff,
+            False,
+        ),
+    ],
+)
+def test_exponential_backoff_with_merge_tree(started_cluster, node, found_in_log):
+    prepare_cluster(False)
+
+    # Executing incorrect mutation.
+    node.query(
+        "ALTER TABLE test_mutations DELETE WHERE x IN (SELECT x  FROM notexist_table) SETTINGS allow_nondeterministic_mutations=1"
```

Comment:
If this setting will be present in `Settings.h`, we could change the setting in runtime via `SETTINGS max_postpone_time=12345`.
Please consider adding a test case changing the setting in runtime